File: LeetCode-2023/2185.统计包含给定前缀的字符串.java
/*
 * @lc app=leetcode.cn id=2185 lang=java
 *
 * [2185] 统计包含给定前缀的字符串
 */

// @lc code=start
class Solution {
    public int prefixCount(String[] words, String pref) {
        int result = 0;
        for(String word : words) {
            // // Solution 1: use indexOf
            // if(word.indexOf(pref) == 0) {
            //     result++;
            // }
            // Solution 2: use startsWith
            if(word.startsWith(pref)) {
                result++;
            }
        }
        return result;
    }
}
// @lc code=end

File: LeetCode-2023/2185.统计包含给定前缀的字符串.py
#
# @lc app=leetcode.cn id=2185 lang=python3
#
# [2185] 统计包含给定前缀的字符串
#

# @lc code=start
class Solution:
    def prefixCount(self, words: List[str], pref: str) -> int:
        return len([x for x in words if x[0:len(pref)] == pref])
        # return sum(x.startswith(pref) for x in words)
# @lc code=end

